test/unit/error.spec.ts

```diff
@@ -292,3 +286,100 @@ test("return from nested finally", () => {
     `;
     expect(util.transpileAndExecute(code)).toBe("finally AB");
 });
+
+test("throw and catch custom error object", () => {
+    util.testFunction`
+      try {
+          throw { x: "Hello error object!" };
+      } catch (error) {
+          return error.x;
+      }
+    `.expectToMatchJsResult();
+});
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as classes",
+    errorType => {
+        util.testFunction`
+            try {
+                throw new ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return \`\${error}\`;
+                } else {
+                    throw Error("This error serves to prevent false positives from .expectNoExecutionError()");
+                }
+            }
+        `
+            .expectNoExecutionError()
+            .expectToMatchJsResult();
+    }
+);
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as functions",
+    errorType => {
+        util.testFunction`
+            try {
+                throw ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return \`\${error}\`;
+                } else {
+                    throw Error("This error serves to prevent false positives from .expectNoExecutionError()");
+                }
+            }
+        `
+            .expectNoExecutionError()
+            .expectToMatchJsResult();
+    }
+);
+
+test("subclass Error", () => {
+    util.testFunction`
+        class MyError extends Error {
+            name: "MyError"
+        }
+        
+        try {
+            throw new MyError();
+        } catch (error) {
+            if (error instanceof Error) {
+                return \'\$error\';
+            } else {
+                throw Error("This error serves to prevent false positives from .expectNoExecutionError()");
+            }
+        }
+    `
+        .expectNoExecutionError()
+        .expectToMatchJsResult();
+});
+
+test.each([["Error", false], ["Error", true], ["RangeError", false], ["MyError", false]] as Array<[string, boolean]>)(
```

Comment:
Inline casts aren't safe, it wouldn't fail even when some element doesn't have a correct type. A better solution would be `as const`, but current jest types don't allow readonly tuples since it supports lower TypeScript versions.

Anyway, I think sourcemap tests should go to `sourcemaps.spec.ts`.

Reply:
It's probably nicer if these were just objects `{error: "RangeError", sourceMapTraceback: false}` anyway. Why not test all combinations by the way?